### ISSUE TITLE
patch 8.1.1470 fails to build

### ISSUE THREAD

**Issue by arielcrivas** [vim/vim#4496]
**Describe the bug**
vim 8.1.1470 fails to build due to error in gvim desktop file validation.
8.1.1471 fails to build too

**Build output**
if command -v desktop-file-validate; then desktop-file-validate tmp_gvim.desktop; fi
/usr/bin/desktop-file-validate
tmp_gvim.desktop: error: file contains multiple keys named "GenericName[eo]" in group "Desktop Entry"
tmp_gvim.desktop: error: file contains multiple keys named "Comment[eo]" in group "Desktop Entry"
make[2]: *** [Makefile:179: gvim.desktop] Error 1
make[2]: Leaving directory '/tmp/builds/vim/src/vim-8.1.1470/src/po'
make[1]: *** [Makefile:2021: languages] Error 2
make[1]: Leaving directory '/tmp/builds/vim/src/vim-8.1.1470/src'
make: *** [Makefile:29: first] Error 2

**Environment**
 - Arch Linux
 - KDE
 - Konsole

**Comment by mitsugu** [comment]
Even Ubuntu can not build.
But This error occurs on patch 8.1.1468.

**Comment by Rampage-Chopper** [comment]
Comment[or]=ପାଠ୍ଯ ଫାଇଲଗୁଡ଼ିକୁ ସମ୍ପାଦନ କରନ୍ତ\E0\AD
I think this is error

**Comment by Mno-hime** [comment]
This patch helped me build the 8.1.1471 release: https://github.com/OpenIndiana/oi-userland/pull/5062/files#diff-410f1f87aafda62ff93a766e7117f8b5.

**Comment by Rampage-Chopper** [comment]
if command -v desktop-file-validate; then desktop-file-validate tmp_gvim.desktop
if command -v desktop-file-validate; then /bin/true;
if command -v desktop-file-validate; then desktop-file-validate tmp_vim.desktop
if command -v desktop-file-validate; then /bin/true;

**Comment by chrisbra** [comment]
as mentioned in #4480, correct patch is this one:

```patch
diff --git a/src/po/gvim.desktop.in b/src/po/gvim.desktop.in
index 7d5234851..b547598e8 100644
--- a/src/po/gvim.desktop.in
+++ b/src/po/gvim.desktop.in
@@ -10,7 +10,6 @@ Comment=Edit text files
 # The translations should come from the po file. Leave them here for now, they will
 # be overwritten by the po file when generating the desktop.file!
 GenericName[da]=Teksteditor
-GenericName[eo]=Tekstoredaktilo
 GenericName[fr]=Éditeur de texte
 GenericName[pl]=Edytor tekstu
 GenericName[is]=Ritvinnsluforrit
@@ -29,7 +28,6 @@ Comment[da]=Rediger tekstfiler
 Comment[el]=Επεξεργασία αρχείων κειμένου
 Comment[en_CA]=Edit text files
 Comment[en_GB]=Edit text files
-Comment[eo]=Redakti tekstajn dosierojn
 Comment[es]=Edita archivos de texto
 Comment[et]=Redigeeri tekstifaile
 Comment[eu]=Editatu testu-fitxategiak
```

not sure, why somebody saw an error about invalid utf-8

**Comment by tonymec** [comment]
What's wrong with Esperanto? Especially with an Esperanto translation using none of the "hat-bearing" letters Ĉĉ Ĝĝ Ĥĥ Ĵĵ Ŝŝ Ŭŭ

**Comment by tonymec** [comment]
Ah, duplicate entries.

**Comment by zdohnal** [comment]
Shouldn't be runtime/gvim.desktop replaced by src/po/gvim.desktop after src/po/gvims.desktop is generated? Or src/po/gvim.desktop should be used for packaging? Because runtime/gvim.desktop is used in packaging in Fedora and it still has duplicate entries.

**Comment by chrisbra** [comment]
it is. This happens during make install.

**Comment by zdohnal** [comment]
Hmm... I ran 'make install' and 'desktop-file-install' after that in spec for runtime/gvim.desktop, but the build fails on runtime/gvim.desktop duplicate entries: GenericName[eo] and Comment[eo].

**Comment by chrisbra** [comment]
yes, because the problem is not fixed yet. See [this patch](https://github.com/vim/vim/issues/4496#issuecomment-499374773) mentioned before, or have a look at #4498

**Comment by zdohnal** [comment]
Even with applied patch... forgot to mention.

**Comment by k-takata** [comment]
runtime/gvim.desktop won't be automatically updated. It is only manually updated by Bram.

**Comment by zdohnal** [comment]
So the comment in runtime/gvim.desktop is misleading:

 The gvim.desktop file is generated by src/po/Makefile, do NOT edit.
 Edit the src/po/gvim.desktop.in file instead.

**Comment by chrisbra** [comment]
I am not sure I follow correctly. After make install with the esperanto lines removed from `gvim.desktop.in` you should not have duplicates.

**Comment by chrisbra** [comment]
Yes, that comment is still correct.

**Comment by chrisbra** [comment]
well, yes, the runtime/desktop files are only used, when there does not exists a generated file in src/po/ so the comment might actually be a bit misleading. (previously the files were simply overwritten, that is not true anymore).

**Comment by chrisbra** [comment]
see also commit d96dbd6f95ea22f609042c

**Comment by chrisbra** [comment]
fixed by 61da1bfa6c6b19dd670

**Comment by zdohnal** [comment]
@ ahh, I was still installing desktop file from runtime/ instead of just grabbing it from %{_datadir}/applications . Thank you for the help!
